shared/Utilities/MSBuildListSplitter.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.BuildTools
+{
+    internal static class MSBuildListSplitter
+    {
+        public static IDictionary<string, string> GetNamedProperties(string[] input)
```

Comment:
Curious why we changed the input type?

Reply:
I learned that MSBuild can bind to `string[]`, providing automatic splitting on `;`. This is necessary if you want to pass in multiple strings, some which may contain semicolons. For example, `$(RestoreSources)` has semicolons, so in Scaffolding I can pass this in as `RestoreSources=$([MSBuild]::Escape($(RestoreSources.Trim())));`